torchgeo/transforms/augmentations.py

```diff
@@ -0,0 +1,115 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""TorchGeo augmentations."""
+
+from typing import Any
+
+import kornia.augmentation as K
+import torch
+from torch import Tensor
+
+
+class SatSlideMix(K.GeometricAugmentationBase2D):
+    """Applies the Sat-SlideMix augmentation to a batch of images and masks.
+
+    Sat-SlideMix rolls (circularly shifts) images along either the height
+    or width axis by a random amount.
+
+    If you use this method in your research, please cite the following paper:
+
+    * https://doi.org/10.1609/aaai.v39i27.35028
+
+    .. versionadded:: 0.8
+    """
+
+    def __init__(
+        self,
+        gamma: int = 1,
+        beta: Tensor | float | tuple[float, float] | list[float] = (0.0, 1.0),
+        p: float = 0.5,
+    ) -> None:
+        """Initialize a new _ExtractPatches instance.
+
+        Args:
+            gamma: The number of augmented samples to create for each
+                input image. The output batch size will be gamma * B.
+            beta: The range of percentage (0.0 to 1.0) of the image
+                dimension (height or width) to shift.
+            p: Probability to apply the augmentation on each sample
+
+        Raises:
+            AssertionError: If `gamma` is not a positive integer.
+        """
+        super().__init__(p=p)
+        assert isinstance(gamma, int) and gamma > 0, 'gamma must be a positive integer'
+        self._param_generator = K.random_generator.PlainUniformGenerator(
```

Comment:
Don't really understand why mypy doesn't like this line, let me know if you want me to do some digging.

Reply:
I think in some cases the attr for param generator is set to None in the base class. I'll take a look again today.